Add OpenBSD/arm64 support for detection of CRC32 and PMULL

brad0 opened the pull request.

ajkr commented on util/crc32c_arm64.cc:
```diff
@@ -46,33 +52,59 @@
 uint32_t crc32c_runtime_check(void) {
-#if !defined(__APPLE__)
+#if defined(__linux__) || defined(__FreeBSD__)
   uint64_t auxv = 0;
 #if defined(ROCKSDB_AUXV_GETAUXVAL_PRESENT)
   auxv = getauxval(AT_HWCAP);
 #elif defined(__FreeBSD__)
   elf_aux_info(AT_HWCAP, &auxv, sizeof(auxv));
 #endif
   return (auxv & HWCAP_CRC32) != 0;
```
Can we move this code to the `#else` branch so it executes in case of non-Apple, non-OpenBSD?

ajkr commented on util/crc32c_arm64.cc:
```diff
@@ -46,33 +52,59 @@
 bool crc32c_pmull_runtime_check(void) {
-#if !defined(__APPLE__)
+#if defined(__linux__) || defined(__FreeBSD__)
   uint64_t auxv = 0;
 #if defined(ROCKSDB_AUXV_GETAUXVAL_PRESENT)
   auxv = getauxval(AT_HWCAP);
 #elif defined(__FreeBSD__)
   elf_aux_info(AT_HWCAP, &auxv, sizeof(auxv));
 #endif
   return (auxv & HWCAP_PMULL) != 0;
```
Same question as above.

ajkr reviewed (changes requested):
The change assumes non-`__APPLE__` implies `__OpenBSD__`, `__FreeBSD__`, or `__linux__`. I may need a reference for that, or the assumption could be removed by addressing the comments on the code.

brad0 commented:
> The change assumes non-`__APPLE__` implies `__OpenBSD__`, `__FreeBSD__`, or `__linux__`. I may need a reference for that, or the assumption could be removed by addressing the comments on the code.

Well originally I had non-APPLE being Linux and FreeBSD. But I changed it to where getauxval() is detected and FreeBSD.

ajkr reviewed (approved):
LGTM, thanks!
